File: webserver/src/main/java/com/nedap/workshop/java/controller/EmployeeController.java
package com.nedap.workshop.java.controller;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import javax.annotation.PostConstruct;
import javax.enterprise.context.RequestScoped;
import javax.inject.Inject;
import javax.inject.Named;

import com.nedap.workshop.java.model.Employee;
import com.nedap.workshop.java.service.EmployeeService;

/**
 * @author paul.vandenberg
 */
@Named(value = "employeeController")
@RequestScoped
public class EmployeeController {

    @Inject
    EmployeeService employeeService;

    @Inject
    Employee employee;

    public List<Employee> getEmployees() {
        return employeeService.findAll();
    }

    public Employee getEmployee() {
        return employee;
    }

    public void save() {
        employeeService.save(employee);
    }
}

File: webserver/src/main/scala/com/nedap/workshop/scala/controller/EmployeeController.scala
package com.nedap.workshop.scala.controller

import javax.enterprise.context.RequestScoped
import javax.inject.{Inject, Named}

import com.nedap.workshop.scala.model.Employee
import com.nedap.workshop.scala.service.EmployeeService

import scala.beans.BeanProperty

/**
  * @author paul.vandenberg
  */
@Named(value = "employeeController")
@RequestScoped
class EmployeeController {
  @Inject
  var employeeService: EmployeeService = _

  @Inject
  @BeanProperty
  var employee: Employee = _

  def getEmployees: java.util.List[Employee] = {
    employeeService.findAll
  }

  def save() {
    employeeService.save(employee)
  }
}
